File: backend/app/schemas/function.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

# Модели для Function (Функция)
class FunctionBase(BaseModel):
    name: str
    code: str
    description: Optional[str] = None
    is_active: bool = True

class FunctionCreate(FunctionBase):
    pass

class FunctionUpdate(BaseModel):
    name: Optional[str] = None
    code: Optional[str] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None

class Function(FunctionBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

# Модели для Section_Function (Связь отдела и функции)
class SectionFunctionBase(BaseModel):
    section_id: int
    function_id: int
    is_primary: bool = True

class SectionFunctionCreate(SectionFunctionBase):
    pass

class SectionFunction(SectionFunctionBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True 

File: frontend/src/types/function.ts
// Типы данных для функций и функциональных назначений

// Тип для функции
export interface Function {
    id: number;
    name: string;
    code: string;
    description?: string | null;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

// Базовый тип для создания/обновления функции
export interface FunctionCreateDTO {
    name: string;
    code: string;
    description?: string;
    is_active?: boolean;
}

// Тип для фильтров при получении функций
export interface FunctionFilters {
    skip?: number;
    limit?: number;
    is_active?: boolean;
}

// Тип для функционального назначения
export interface FunctionalAssignment {
    id: number;
    position_id: number;
    function_id: number;
    percentage: number;
    is_primary: boolean;
    created_at: string;
    updated_at: string;
}

// Базовый тип для создания/обновления функционального назначения
export interface FunctionalAssignmentCreateDTO {
    position_id: number;
    function_id: number;
    percentage?: number;
    is_primary?: boolean;
}

// Тип для фильтров при получении функциональных назначений
export interface FunctionalAssignmentFilters {
    skip?: number;
    limit?: number;
    position_id?: number;
    function_id?: number;
    is_primary?: boolean;
} 